packages/airnode-protocol/contracts/rrp/AirnodeRrpV0DryRun.sol

```diff
@@ -0,0 +1,110 @@
+        bytes data
+    );
+
+    // It is virtually impossible for a contract to be deployed at this address
+    address private constant ADDRESS_WITH_NO_BYTECODE = address(uint160(uint256(keccak256("Address with no bytecode"))));
+
+    /// @dev This mapping is kept as it is in AirnodeRrpV0 to closely simulate
+    /// the fulfillment. All of its keys will map to zero values.
+    mapping(bytes32 => bytes32) private requestIdToFulfillmentParameters;
+
+    // solhint-disable-next-line payable-fallback
+    fallback() external { }
+
+    /// @notice Used by Airnode to estimate the gas amount needed to fulfill
+    /// the request (excluding the external call). Do not call this function,
+    /// as it will have no practical effect.
+    /// @dev Refer to AirnodeRrpV0's `fulfill()` for more information
+    /// @param requestId Request ID
+    /// @param airnode Airnode address
+    /// @param data Fulfillment data
+    /// @param fulfillAddress Address that will be called to fulfill
+    /// @param fulfillFunctionId Signature of the function that will be called
+    /// to fulfill
+    /// @return callSuccess If the fulfillment call succeeded
+    /// @return callData Data returned by the fulfillment call (if there is
+    /// any)
+    function fulfill(
+        bytes32 requestId,
+        address airnode,
+        address fulfillAddress,
+        bytes4 fulfillFunctionId,
+        bytes calldata data,
+        bytes calldata signature
+    ) external returns (bool callSuccess, bytes memory callData) {
+        // The line below is kept the same, except that the condition is
+        // reversed to ensure that it never reverts. All
+        // `requestIdToFulfillmentParameters` values are zero and virtually any
+        // `keccak256()` output will not be equal to that.
+        require(
+            keccak256(
+                abi.encodePacked(
+                    airnode,
+                    msg.sender,
+                    fulfillAddress,
+                    fulfillFunctionId
+                )
+            ) != requestIdToFulfillmentParameters[requestId],
+            "Dummy revert string"
+        );
+        // The line below does not need to be modified
+        require(
+            (
+                keccak256(abi.encodePacked(requestId, data))
+                    .toEthSignedMessageHash()
+            ).recover(signature) == airnode,
+            "Invalid signature"
+        );
+        // We cannot call `fulfillAddress` below because (1) we do not want
+        // this function to actually fulfill the request (2) the fulfill
+        // function will be behind an `onlyAirnodeRrp` modifier and will reject
+        // the calls from AirnodeRrpV0DryRun.
+        // Instead, we call an address that we know to not contain any
+        // bytecode, which means this call will not revert or spend extra gas.
+        (callSuccess, callData) = ADDRESS_WITH_NO_BYTECODE.call( // solhint-disable-line avoid-low-level-calls
```

Comment:
I didn't want to use `address(0)` since some chains may have a contract at that address for whatever reason

Reply:
I was wondering just that 👍🏻 
We can also safely assume that `airnode` and `msg.sender` are EOAs without bytecode, wouldn't those be an alternative as well?

Reply:
I think you can have `msg.sender` be a contract (if you can get the requester and the Airnode to play along), but I see how `airnode` is guaranteed to be an EOA. Do you think that we should use that instead?

Reply:
> would allow one to build trustless sponsor wallets using https://eips.ethereum.org/EIPS/eip-3074 in the future

🤯 

> Do you think that we should use that instead?

I don't have a strong opinion about it. It is very unlikely that ADDRESS_WITH_NO_BYTECODE will collide with a smart contract address anyways so I think it's fine too.